org.eclipse.paho.android.service/src/main/java/org/eclipse/paho/android/service/MqttAndroidClient.java

```diff
@@ -1526,6 +1528,28 @@ private void simpleAction(IMqttToken token, Bundle data) {
 		}
 	}
 
+	private void subscribeResultAction(IMqttToken token, Bundle data) {
+		if (token != null) {
+			int[] grantedQos = data.getIntArray(MqttServiceConstants.GRANTED_QOS);
+			String[] topics = token.getTopics();
+			List<String> successTopics = new ArrayList<>();
+			List<String> failedTopics = new ArrayList<>();
+			if (grantedQos != null) {
+				for (int i = 0; i < topics.length; i++) {

```

Comment:
can use foreach here or can make topics.length as constant so that it is not recomputed
https://stackoverflow.com/questions/256859/is-there-a-performance-difference-between-a-for-loop-and-a-for-each-loop

Reply:
Have taken `topics.length` in a separate variable. Index is required to get granted qos hence foreach loop is not feasible